src/SshNetTests/SftpTests.cs

```diff
@@ -4426,6 +4447,400 @@ public void Sftp_SftpFileStream_Seek_BeyondEndOfFile()
+                        Assert.AreEqual(readBuffer.Length, fs.Read(readBuffer, 0, readBuffer.Length));
+                        Assert.IsTrue(writeBuffer.IsEqualTo(readBuffer));
+
+                        Assert.AreEqual(-1, fs.ReadByte());
+                    }
+
+                    client.DeleteFile(remoteFile);
+
+                    // create single-byte file
+                    using (var fs = client.OpenWrite(remoteFile))
+                    {
+                        fs.WriteByte(0x04);
+                    }
+
+                    // seek beyond EOF and beyond buffer size
+                    // write less bytes than buffer size
+                    seekOffset = 700L;
+                    writeBuffer = GenerateRandom(4);
+
+                    using (var fs = client.OpenWrite(remoteFile))
+                    {
+                        var newPosition = fs.Seek(seekOffset, SeekOrigin.End);
+
+                        Assert.AreEqual(1 + seekOffset, newPosition);
+                        Assert.AreEqual(newPosition, fs.Position);
+
+                        fs.Write(writeBuffer, 0, writeBuffer.Length);
+                    }
+
+                    using (var fs = client.OpenRead(remoteFile))
+                    {
+                        Assert.AreEqual(1 + seekOffset + writeBuffer.Length, fs.Length);
+                        Assert.AreEqual(0x04, fs.ReadByte());
+
+                        var emptyBuffer = new byte[seekOffset];
+                        Assert.AreEqual(emptyBuffer.Length, fs.Read(emptyBuffer, 0, emptyBuffer.Length));
+                        Assert.IsTrue(new byte[emptyBuffer.Length].IsEqualTo(emptyBuffer));
+
+                        var readBuffer = new byte[writeBuffer.Length];
+                        Assert.AreEqual(readBuffer.Length, fs.Read(readBuffer, 0, readBuffer.Length));
+                        Assert.IsTrue(writeBuffer.IsEqualTo(readBuffer));
+
+                        Assert.AreEqual(-1, fs.ReadByte());
+                    }
+
+                    client.DeleteFile(remoteFile);
+
+                    // create single-byte file
+                    using (var fs = client.OpenWrite(remoteFile))
+                    {
+                        fs.WriteByte(0x04);
+                    }
+
+                    // seek beyond EOF but not beyond buffer size
+                    // write more bytes than buffer size
+                    seekOffset = 3L;
+                    writeBuffer = GenerateRandom(600);
+
+                    using (var fs = client.OpenWrite(remoteFile))
+                    {
+                        var newPosition = fs.Seek(seekOffset, SeekOrigin.End);
+
+                        Assert.AreEqual(1 + seekOffset, newPosition);
+                        Assert.AreEqual(newPosition, fs.Position);
+
+                        fs.Write(writeBuffer, 0, writeBuffer.Length);
+                    }
+
+                    using (var fs = client.OpenRead(remoteFile))
```

Comment:
This validation code is repeated several times. Should it be moved to a parametized helper method?

Reply:
I prefer to keep this as is to avoid having to navigate too much between method to find out what is actually being tested.
OK?

Reply:
I'm not too fond of how verbose and repetitive the current approach is, but it is not a showstopper.